libs/clients/health-directorate/src/lib/clients/vaccinations/vaccinations.service.ts

```diff
@@ -0,0 +1,49 @@
+import { AuthMiddleware, Auth } from '@island.is/auth-nest-tools'
+import { handle404 } from '@island.is/clients/middlewares'
+import { Injectable } from '@nestjs/common'
+import {
+  DiseaseVaccinationDto,
+  MeVaccinationControllerGetVaccinationForDiseaseRequest,
+  MeVaccinationControllerGetVaccinationsForDiseasesRequest,
+  MeVaccinationsApi,
+  VaccinationDto,
+} from './gen/fetch'
+
+@Injectable()
+export class HealthDirectorateVaccinationsService {
+  constructor(private readonly vaccinationsApi: MeVaccinationsApi) {}
+
+  private vaccinationsApiWithAuth(auth: Auth) {
+    return this.vaccinationsApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  public async getVaccinations(
+    auth: Auth,
+    input: MeVaccinationControllerGetVaccinationsForDiseasesRequest,
+  ): Promise<Array<DiseaseVaccinationDto> | null> {
+    const vaccines = await this.vaccinationsApiWithAuth(auth)
+      .meVaccinationControllerGetVaccinationsForDiseases(input)
+      .catch(handle404)
+
+    if (!vaccines) {
+      return null
```

Comment:
Do we need logging here and on l:44 if there's a null response?

Something like this, maybe with more info:

```
      this.logger.warn('No vaccines returned', {
        category: LOG_CATEGORY,
      })
```

Reply:
Updated - also adder warning in organ donation as they should not be null either :) 
Can you re-review 😇 ?